### PR TITLE
Add '.cppm' files as C++ sources.

### PR DESCRIPTION
**Pull request by robert-andrzejuk**
It is becoming common to use "cppm" or "ixx" as extension for C++ modules. This is standard C++ code.
"ixx" is already on the list.

Reference: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5926

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?q=path%3A*.cppm&type=code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/kaladron/galactic-bloodshed/blob/651676152d2f21b380ef1c3e109edc0d33e622df/gb/gblib.cppm#L4
    - Sample license(s):
      - https://github.com/kaladron/galactic-bloodshed/blob/master/LICENSE 
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by robert-andrzejuk** [comment]
@lildude 
I edited my post to add the link to the github query: https://github.com/search?q=path%3A*.cppm&type=code
Is that all that is required?

**Comment by lildude** [comment]
Nope. You have missed the point in the CONTRIBUTING.md file about preferring real world samples. You've clearly added a "hello world" sample which is never considered real world code. Please add a real world sample and update the OP to link to the source and state the license of the sample.

**Comment by robert-andrzejuk** [comment]
Ok. Thanks for you help.

I copied a file from a repository. And attached it in samples/C++ section.
Also updated my original post (OP). (OP was an unknown abbreviation to me, till now.)

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -740,6 +740,7 @@ C++:
   - cpp
   extensions:
   - ".cpp"
+  - ".cppm"
   - ".c++"
   - ".cc"
   - ".cp"
```
Comment (lildude) [inline lib/linguist/languages.yml:746]:
```suggestion
  - ".c++"
  - ".cc"
  - ".cp"
  - ".cppm"
```

**Review by lildude — APPROVED** [review]

**Pull request merged**